test/spock/src/test/groovy/MessageServiceSpec.groovy

```diff
@@ -6,24 +6,24 @@ class MessageServiceSpec extends Specification {
 
     def 'Get message'() {
         URLConnection con = (URLConnection) new URL("http://postman-echo.com/headers").openConnection()
-        con.requestMethod = "GET"
-        def bytes = con.inputStream.readAllBytes()
+        def bytes = con.inputStream.bytes
+        String expectedContent = URLEncoder.encode("Get message", "UTF-8")
         expect: 'Should return the correct message'
         println 'Should return the correct message'
-        new String(bytes).contains("Get message")
+        new String(bytes).contains(expectedContent)
     }
 
     def "numbers to the power of two"(int a, int b, int c) {
         URLConnection con = (URLConnection) new URL("http://postman-echo.com/headers").openConnection()
-        con.requestMethod = "GET"
-        def bytes = con.inputStream.readAllBytes()
+        def bytes = con.inputStream.bytes
+        String expectedContent = URLEncoder.encode("numbers to the power of two", "UTF-8")
```

Comment:
URLEncoder.encode("smth", "UTF-8") duplicates 4 times, it should be in the static fun